### PR TITLE
sql: properly scope cluster_contended_indexes

### LINKED ISSUE #72700: 21.2 Contention view returns incorrect results
We released contention views (contended indexes) in 21.2: https://github.com/cockroachdb/cockroach/issues/61507. An issue has been reported where this view would be returning wrong results. 

Doc issue outlining the issue: https://github.com/cockroachdb/docs/issues/12269

We need to correlate the contention_event's table id with table_index's descriptor id. Feedback also mentioned we should have the results DESC order.

Two options:

SELECT DISTINCT t.database_name, t.schema_name, t.name, ti.index_name, ce.num_contention_events
FROM crdb_internal.cluster_contention_events ce
INNER JOIN crdb_internal.table_indexes ti
  ON ce.index_id = ti.index_id
AND ce.table_id = ti.descriptor_id
INNER JOIN crdb_internal.tables t
  ON ce.table_id = t.table_id 
ORDER BY ce.num_contention_events DESC;

OR

SELECT DISTINCT database_name, schema_name, name, index_name, num_contention_events
FROM crdb_internal.cluster_contention_events, crdb_internal.tables, crdb_internal.table_indexes
WHERE (crdb_internal.cluster_contention_events.index_id = crdb_internal.table_indexes.index_id AND crdb_internal.cluster_contention_events.table_id = crdb_internal.table_indexes.descriptor_id)
AND (crdb_internal.cluster_contention_events.table_id = crdb_internal.tables.table_id)
ORDER BY num_contention_events DESC;

### PR DESCRIPTION
**Pull request by matthewtodd**
Fixes #72700.

We released contention views in 21.2, later discovering we mis-scoped
the query backing `crdb_internal.cluster_contended_indexes`.

This change corrects the scoping and also sorts the results usefully.

Release note (bug fix): the query backing
crdb_internal.cluster_contended_indexes improperly assumed that index
ids were unique across the database. This change adds the proper scoping
by table descriptor id, truing up the contents of that view.

### CODE REVIEW

**Review by maryliag — APPROVED** [review]
:lgtm: 

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72780#-:-ModYLIO-a8uKC2t-fTq:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kevin-v-ngo)

File: pkg/sql/crdb_internal.go [hunk 1]
```diff
@@ -1932,10 +1932,18 @@ CREATE VIEW crdb_internal.cluster_contended_indexes (
...
+      = crdb_internal.table_indexes.index_id
+      AND crdb_internal.cluster_contention_events.table_id
+        = crdb_internal.table_indexes.descriptor_id
+    )
+    AND (
+        crdb_internal.cluster_contention_events.table_id
+        = crdb_internal.tables.table_id
+      )
+  ORDER BY
+    num_contention_events DESC
```
Comment (Azhng) [inline pkg/sql/crdb_internal.go:1946]: Hmm Sorry that i'm a bit late to this. Are these parenthesis around the `WHERE` clause necessary ?

Comment (matthewtodd) [reply pkg/sql/crdb_internal.go:1946]: Oops, no. #72929.

**Pull request merged**